packages/autorest.typescript/src/generators/static/packageFileGenerator.ts

```diff
@@ -194,7 +194,8 @@ function regularAutorestPackage(
     packageInfo.devDependencies["@azure/identity"] = "^2.0.1";
     packageInfo.devDependencies["@azure-tools/test-recorder"] = "^3.0.0";
     packageInfo.devDependencies["@azure-tools/test-credential"] = "^1.0.0";
-    packageInfo.devDependencies["mocha"] = "^7.1.1";
+    packageInfo.devDependencies["mocha"] = "^10.0.0";
+    packageInfo.devDependencies["esm"] = "^3.2.18";
```

Comment:
looks like this is missing @types/mocha ?

Reply:
The old version doesn't include @types/mocha but I probably should add it too.

Reply:
I was assuming that some packages might be fine without it (JS repo just upgraded to mocha v10 and some generated packages there may be missing @types/mocha.  But adding it shouldn't hurt.

Reply:
Thank you :)

Reply:
fixed